### ISSUE TITLE
[CA1066] Does not look for implemented interfaces recursivelly

### ISSUE THREAD

**Issue by skarllot** [dotnet/roslyn-analyzers#1914]
#### Analyzer package

[Microsoft.CodeQuality.Analyzers](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers)

#### Package Version

v2.6.2

#### Diagnostic ID

CA1066 (No documentation)

#### Repro steps

1. Create a generic interface:
```csharp
public interface IValueObject<T> : IComparable<T>, IEquatable<T> {}
```
2. Create a type that implements the interface above:
```csharp
public readonly struct Meters : IValueObject<Address>
{
    private readonly int value;

    public override bool Equals(object obj) => obj is Meters other && Equals(other);

    public bool Equals(Meters other) => value == other.value;

    public override int GetHashCode() => value;

    // ...
}
```

#### Expected behavior
Should not warn about no implementation of IEquatable`1 interface.

#### Actual behavior
It is warning about missing implementation of IEquatable`1 interface.
